front/src/applications/operationalStudies/components/SimulationResults/SpaceTimeChart/drawTrain.ts

```diff
@@ -46,26 +55,29 @@ export default function drawTrain(
    *
    * @param {int} offset
    */
-  const applyTrainCurveTranslation = (offset) => {
+  const applyTrainCurveTranslation = (offset: number) => {
     const translation = rotate ? `0,${offset}` : `${offset},0`;
     d3.select(`#${groupID}`).attr('transform', `translate(${translation})`);
     const releventLine = rotate ? '#horizontal-line' : '#vertical-line';
     d3.select(releventLine).attr('transform', `translate(${translation})`);
   };
 
-  let debounceTimeoutId;
+  let debounceTimeoutId: number;
 
-  function debounceUpdateDepartureArrivalTimes(computedDepartureArrivalTimes, interval) {
+  function debounceUpdateDepartureArrivalTimes(
+    computedDepartureArrivalTimes: TrainsWithArrivalAndDepartureTimes[],
+    interval: number
+  ) {
     clearTimeout(debounceTimeoutId);
-    debounceTimeoutId = setTimeout(() => {
+    debounceTimeoutId = window.setTimeout(() => {
```

Comment:
do you really need this?

Reply:
Yes, because TS consider that it can be Nodejs setTimeout which return an object `Timeout` while `debounceTimeoutId` is a number